### PR TITLE
[SPARK-15331] [SQL] Disallow All the Unsupported CLI Commands

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

The JIRAs under https://issues.apache.org/jira/browse/SPARK-14118 remove the support of the DDL/CLI commands that are not natively supported by Spark. However, in Spark-SQL CLI, we still delegate the CLI commands to Hive command processors. To be consistent, we should also remove the related command supports in Spark SQL CLI.

Also, we need to improve Parser to issue better error exception messages for these unsupported operations. 

```
- DFS,
- RELOAD FUNCTION
- DELETE FILE/FILES/JAR/JARS/ARCHIVE/ARCHIVES
- SOURCE
```
#### How was this patch tested?

Added a test case in CliSuite.

### CODE REVIEW

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -172,6 +172,10 @@ unsupportedHiveNativeCommands
     | kw1=ROLLBACK
     | kw1=DFS
     | kw1=DELETE kw2=FROM
+    | kw1=RELOAD kw2=FUNCTION
+    | kw1=DELETE kw2=(FILE | FILES | JAR | JARS | ARCHIVE | ARCHIVES)
```
Comment (hvanhovell) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Could you perhaps do this without adding the `FILE`, `FILES`, `JAR`, `JARS`, `ARCHIVE` & `ARCHIVES` keywords? This makes it consistent with the `ADD <resource>` command.

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala [hunk 2]
```diff
@@ -234,6 +234,13 @@ class CliSuite extends SparkFunSuite with BeforeAndAfterAll with Logging {
     )
   }
 
+  test("unsupported operations") {
```
Comment (hvanhovell) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: I suppose there is not a time efficient way to test all commands?

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala [hunk 3]
```diff
@@ -312,7 +311,10 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
       if (proc != null) {
         // scalastyle:off println
         if (proc.isInstanceOf[Driver] || proc.isInstanceOf[SetProcessor] ||
-          proc.isInstanceOf[AddResourceProcessor]) {
+            proc.isInstanceOf[AddResourceProcessor] || proc.isInstanceOf[DeleteResourceProcessor] ||
```
Comment (hvanhovell) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala]: Pattern match?

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 (hunk 1 above)
Comment (gatorsmile) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Sure, will do it.

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala (hunk 3 above)
Comment (gatorsmile) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala]: Yeah, will do it.

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala (hunk 2 above)
Comment (gatorsmile) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: Yeah, this is the only we can check if CLI will reject the command. This test suite is slow but we can do it for all the unsupported operations.

Comment (gatorsmile) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala]: @hvanhovell The latest changes added the test cases for the unsupported operations.

**Comment by gatorsmile** [comment]
@hvanhovell @yhuai This is also ready to review. Thanks!

**Pull request closed without merging**